### ISSUE TITLE
Seven character numeric strings are being falsely interpreted as dates

### ISSUE THREAD

**Issue by hoenth** [andyjeffries/flexirest#7]
When a value includes a string like "1266126", the regex on line 27 in Flexirest::Base is falsely interpreting it as a date, and parsing it. 

Could the regex be more explicit in checking date values, using a regex like the following:
````regex
^((19|20)\d\d[- \/.](0[1-9]|1[012]|[1-9])[- \/.](0[1-9]|[12][0-9]|3[01]|[1-9]))|((0[1-9]|1[012]|[1-9])[- \/.](0[1-9]|[12][0-9]|3[01]|[1-9])[- \/.](19|20)\d\d)$
````
This requires that the string be in US or EU style, with delimiters.

**Comment by andyjeffries** [comment]
I agree with your suggestion.  Have you raised a PR with this? Your anchoring of the regex will also fix https://github.com/whichdigital/active-rest-client/pull/112.

If you haven't raised a PR (and aren't planning on it), I'll fix it sometime in early January when I'm back working.

**Comment by hoenth** [comment]
I will issue a PR later this week. I am not a regex expert by any stretched
and had some difficulty getting the regex for the date time to work
properly.  Let me know if you have a better suggestion when you take a look
at the pr.

Ton

sent from phone, please excuse typos and brevity.

**Comment by andyjeffries** [comment]
I've merged your PR and released it as v1.2.10
